lib.js

```diff
@@ -88,29 +89,52 @@ module.exports = ({
           authorName,
           authorEmail,
           license,
+          generateExample,
         };
 
-        const filename = path.join(name, template.name(args));
-        var baseDir = filename.split(path.basename(filename))[0];
-        
-        return createFolder(baseDir).then(() =>
-          createFile(filename, template.content(args))
-        );
+        return renderTemplate(name, template, templateArgs);
       }));
     })
     .then(() => {
+      // Generate the example if necessary
       if (!generateExample) {
         return Promise.resolve();
       }
-      // Adds and links the created library project
-      const pathExampleApp = `./${name}/example`;
-      const options = { cwd: pathExampleApp, stdio:'inherit'};
-      try {
-        execSync('yarn add file:../', options);
-      } catch (e) {
-        execSync('npm install ../', options);
-        execSync('npm install', options);
-      }
-      execSync('react-native link', options);
+
+      const initExampleOptions = { cwd: `./${name}`, stdio: 'inherit' };
+      return exec('react-native init example', initExampleOptions)
+        .then(() => {
+          // Execute the example template
+          const exampleTemplates = require('./templates/example');
+          return Promise.all(
+            exampleTemplates.map((template) => {
+              return renderTemplate(name, template);
+            })
+          );
+        })
+        .then(() => {
+          // Adds and link the new library
+          return new Promise((resolve, reject) => {
+            // Add postinstall script to example package.json
+            const pathExampleApp = `./${name}/example`;
+            const moduleName = `${modulePrefix}-${paramCase(name)}`;
+            npmAddScriptSync(`${pathExampleApp}/package.json`, {
+              key: 'postinstall',
+              value: `node ../scripts/examples_postinstall.js node_modules/${moduleName}`
+            });
+
+            // Add and link the new library
+            const addLinkLibraryOptions = { cwd: pathExampleApp, stdio: 'inherit' };
+            try {
+              execSync('yarn add file:../', addLinkLibraryOptions);
+            } catch (e) {
+              execSync('npm install ../', addLinkLibraryOptions);
+              execSync('npm install', addLinkLibraryOptions);
```

Comment:
Did you mean to leave this one in here?

Reply:
I didn't know why it was in there in the first place. Look on the left side except renaming the options variable the logic didn't change. Do you recall why we have that in there twice?